playbooks/molecule/resources/xnat/verify.yml

```diff
@@ -16,3 +16,73 @@
           - response.status == 200
           - response.server == "nginx"
           - response.content is search('MIRSG XNAT')
+
+- name: Verify correct Docker version is installed
+  hosts: xnat_cserv
+  tasks:
+    - name: Get Docker server version
+      ansible.builtin.command:
+        cmd: "{% raw %}docker version -f '{{.Server.Version}}'{% endraw %}"
```

Comment:
the `{% raw %}` block is necessary to prevent Ansible from trying to do variable substitution on `'{{.Server.Version}}'`

Reply:
How does this verify the server version?

Locally, I have upon `docker version`

```sh
Client: Docker Engine - Community
 Version:           26.1.3
 API version:       1.44 (downgraded from 1.45)
 Go version:        go1.22.3
 Git commit:        b72abbb6f0
 Built:             Thu May 16 07:47:24 2024
 OS/Arch:           darwin/arm64
 Context:           orbstack

Server: Docker Engine - Community
 Engine:
  Version:          25.0.5
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       e63daec
  Built:            Tue Mar 19 15:05:27 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.7.13
  GitCommit:        7c3aca7a610df76212171d200ca3811ff6096eb8
 runc:
  Version:          1.1.12
  GitCommit:        51d5e94601ceffbbd85688df1c928ecccbfa4685
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

When I run `docker version -f 25.0.5` or `docker version -f 26.1.3` they both return the version number.

Reply:
> How does this verify the server version?
> When I run docker version -f 25.0.5 or docker version -f 26.1.3 they both return the version number.

If you run `docker version -f '{{.Server.Version}}'` it will return `25.0.5`, and `docker version -f '{{.Client.Version}}'` will return `26.1.3` - it's using the [`--format` option](https://docs.docker.com/reference/cli/docker/version/#examples) to filter the content returned by `docker version`

Reply:
Ah, sorry, I misunderstood the syntax. Have verified this locally.

Reply:
no worries - that's what the `raw` block is for, to ensure the raw string `'{{.Server.Version}}'` is passed to the `docker version` command rather than trying to substitute in the version first and then pass that

Reply:
I got that bit, I just didn't then follow through to realise that's what I need to type in the terminal 😆